tools/remap/convert_remap_actions_to_10x

```diff
@@ -0,0 +1,172 @@
+        # Find the largest <num> in input_dir/input_base.action_to_add.bak.<num>.
+        max_num = 0
+        for file in os.listdir(input_dir):
+            if file.startswith(f'{input_base}.action_to_add.bak.'):
+                try:
+                    num = int(file.split('.')[-1])
+                except ValueError:
+                    # The suffix is not a number. Ignore this file.
+                    continue
+                max_num = max(max_num, num)
+        copy_path = os.path.join(input_dir, f'{input_base}.action_to_add.bak.{max_num + 1}')
+    shutil.copyfile(input, copy_path)
+    return copy_path
+
+
+def prepare_files(input: str, output: Optional[str], no_backup: bool) -> Tuple[str, str]:
+    '''Prepare the input and output files.
+
+    :param input: The input file path.
+    :param output: The output file path.
+    :param no_backup: Do not create a backup of the input file when <input> is modified.
+
+    :return: A tuple containing the input and output filenames, in that order.
+    The input file will contain the content from which to copy, the output file
+    will reference a file whose content should be overridden. The caller should
+    delete the input filename if @a no_backup is true. Otherwise the input
+    filename will be the path to the backup file.
+    '''
+    if not os.path.exists(input):
+        raise FileNotFoundError(f'Input file does not exist: {input}')
+
+    if input == output:
+        input_path = copy_input_file(input, no_backup)
+        output_path = input
+    elif output is None:
+        input_path = copy_input_file(input, no_backup)
+        output_path = input
+    else:
+        input_path = input
+        output_path = output
+    return input_path, output_path
+
+
+def parse_args() -> argparse.Namespace:
+    '''Parse command line arguments.
+
+    :return: The parsed arguments.
+    '''
+    parser = argparse.ArgumentParser(description=help)
+    parser.add_argument('input', help='The input remap.config file to modify.')
+    parser.add_argument(
+        '-o',
+        '--output',
+        help='The output remap.config file. By default, output is written to <input> and a backup file is generated.')
+    parser.add_argument(
+        '-n', '--no-backup', action='store_true', help='Do not create a backup of the input file when <input> is modified.')
+    return parser.parse_args()
+
+
+def main() -> int:
+    '''Convert the input remap.config file.
+
+    :return: The process exit code.
+    '''
+
+    args = parse_args()
+    input_path, output_path = prepare_files(args.input, args.output, args.no_backup)
+
+    with open(input_path, 'r') as fd_in, open(output_path, 'w') as fd_out:
+        for input_line in fd_in:
+            output_line = convert_line(input_line)
+            fd_out.write(output_line)
+    if args.no_backup:
+        os.remove(input_path)
+    return 0
+
+
+if __name__ == '__main__':
+    import doctest
+    doctest.testmod()
```

Comment:
A bit messy to always run doctest. The user has no way to disable this functionality. :/

Reply:
I'd feel comfortable removing it if `python3 -m doctest convert_remap_actions_to_10x` worked. I've never run in this way and I seem to not be able to make it work:

```shell
╰─➤  cat d
#!/usr/bin/env python3

def add_one(i: int) -> int:
    '''Add one to the input.

    :param i: The input to increment.
    :return: One ore than @a i.
    :example:
    >>> add_one(2)
    3
    >>> add_one(0)
    1
    '''

    return i + 1

n = add_one(0)
print(n)
╭─[31/07/24 6:14:53] bneradt@lima-rancher-desktop  /tmp 
╰─➤  python3 -m doctest -v d
Trying:
    add_one(2)
Expecting:
    3
**********************************************************************
File "d", line 9, in d
Failed example:
    add_one(2)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib64/python3.12/doctest.py", line 1368, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest d[0]>", line 1, in <module>
        add_one(2)
        ^^^^^^^
    NameError: name 'add_one' is not defined
Trying:
    add_one(0)
Expecting:
    1
    '''
**********************************************************************
File "d", line 11, in d
Failed example:
    add_one(0)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib64/python3.12/doctest.py", line 1368, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest d[1]>", line 1, in <module>
        add_one(0)
        ^^^^^^^
    NameError: name 'add_one' is not defined
**********************************************************************
1 items had failures:
   2 of   2 in d
2 tests in 1 items.
0 passed and 2 failed.
***Test Failed*** 2 failures.
```

Reply:
What about `python3 -m doctest -v convert_remap_actions_to_10x`?

Reply:
Looks like that just adds a verbose flag.

Reply:
I'm fine with leaving it.

Reply:
Just to record it, you found that this is due to the script not having a .py extension. I'd prefer to leave the doctest in the code than to add a .py extension. Scripts shouldn't need to advertise their implementation.